cmd/zpool/zpool_main.c

```diff
@@ -676,6 +676,20 @@ zpool_do_add(int argc, char **argv)
 		return (1);
 	}
 
+	/* unless manually specified use "ashift" pool property for new vdevs */
+	if (!nvlist_exists(props, ZPOOL_CONFIG_ASHIFT)) {
```

Comment:
> needs to be able to be changed

I don't understand why it _needs_ to be: if the property wasn't set at pool creation and the user needs to override the default value they can always `zpool add -o ashift=<n>`.

My fear is that making it editable can potentially create situations were the user, especially now that ZFS is reaching a broader audience, believes that changing the value will magically fix every problem caused by mixing storage with different block sizes. On the other hand leaving the property "onetime" is not doing any real harm.